### PR TITLE
Privacy Considerations refresh

### LINKED ISSUE #85: Fingerprinting via machine-specific artifacts
Apropos of webmachinelearning/webnn#3 and webmachinelearning/ethical-webmachinelearning#22, an efficient matmul implementation can be fingerprinted to determine hardware capabilities.  

On pre-VNNI Intel, the only efficient way to implement 8-bit multiplication is via `pmaddubsw` that produces a 16-bit result summed horizontally with saturation.  I can construct matrices that test for this saturation, which indicates a pre-VNNI Intel CPU.  Whereas ARM and NVidia implement signed * signed to 32-bit.  

Saturating addition, which should be used for accuracy lest you generate large sign errors, can be used to infer the order of operations.  So `vpdpbusds` saturation tells me what order the `matmul` ran in.  

The slowdown from using AVX512 instructions is likely detectable with timing.  

In floats one can also infer order of operations from rounding.  This would reveal the SIMD length and possibly variations in the compiler used to build the user agent.  A cache-efficient matmul implementation reveals cache sizes via floating point order of operations.

### LINKED ISSUE #175: Related WebGPU Security and Privacy Considerations
I'm opening this issue to keep track of related WebGPU Security and Privacy Considerations that may apply in the context of the WebNN API, and could be explicitly referenced:

 - WebGPU: https://gpuweb.github.io/gpuweb/#malicious-use

Here's an opportunity to all privacy and security experts and interested folks to review the WebGPU Security and Privacy Considerations and propose whether we should explicitly reference some specific considerations in this Web neural Network API spec.

Currently, we have added the following statement to the Privacy Considerations:

>Implementers of this API are expected to be familiar with the [WebGPU Privacy Considerations](https://gpuweb.github.io/gpuweb/#security-privacy).

See https://webmachinelearning.github.io/webnn/#security and https://webmachinelearning.github.io/webnn/#privacy

### PR DESCRIPTION
**Pull request by anssiko**
Add the following to Privacy Considerations:

- Machine-specific artifacts
- Device selection
- Future device types

Fix #85
Fix #175
Related #169

PTAL @krgovind @RafaelCintron

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/pull/271.html" title="Last updated on Jun 16, 2022, 4:19 PM UTC (5185b9f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/271/e9b9e2c...5185b9f.html" title="Last updated on Jun 16, 2022, 4:19 PM UTC (5185b9f)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -436,6 +436,12 @@ Note: The group is <a href="https://github.com/webmachinelearning/webnn/issues/8
...
+
+The WebNN API defines two developer-settable preferences to help inform [[#programming-model-device-selection]] and allow the implementation better select the most appropriate underlying execution device for the workload. [=Device type=] preference indicates the kind of device and is either "cpu" or "gpu". If this preference cannot be satisfied, an {{OperationError}} exception is thrown, thus this preference can in some cases add two bits of entropy to the fingerprint. [=Power preference=] indicates preference as related to the power consumption and is considered a hint only and as such does not increase entropy of the fingerprint.
```
Comment (huningxin) [inline index.bs:441]: a nit: [=Device type=] is not a preference but a normative type in current spec.

Comment (anssiko) [reply index.bs]: Thanks @huningxin, proposed fix in https://github.com/webmachinelearning/webnn/pull/271/commits/5185b9fbc524a8a1409d1a691676dedfd8252644

Comment (huningxin) [reply index.bs]: looks good, thanks @anssiko !

**Review by huningxin — APPROVED** [review]

**Comment by anssiko** [comment]
Thanks @huningxin for your review and comments.

I'd like to get additional review from @wchao1115 and @RafaelCintron prior to submitting our spec for Privacy review refresh.

**Comment by anssiko** [comment]
Merging per https://www.w3.org/2022/06/30-webmachinelearning-minutes.html#r01

**Pull request merged**